survey_partner_representative/models/survey_representative_mixin.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2024 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class SurveyRepresentativeMixin(models.AbstractModel):
+    """This abstract is handy when we have a generated record that is linked to a
+    user input. This way, we can compute the represetative partner right away"""
+
+    _name = "survey.representative.mixin"
+    _description = "Use this mixin in models that have a linked survey input"
+
+    survey_user_input_id = fields.Many2one(
+        comodel_name="survey.user_input", readonly=True
+    )
+    survey_representative_partner_id = fields.Many2one(
+        comodel_name="res.partner",
+        compute="_compute_survey_representative_partner_id",
+        store=True,
+    )
+
+    @api.depends("survey_user_input_id")
+    def _compute_survey_representative_partner_id(self):
+        for record in self:
+            record.survey_representative_partner_id = (
```

Comment:
I suppose this one can't be a related due to the initialization, so put a comment about that for not being tempted in the future to change it. Not being covered in tests may mean dead code?

Reply:
The idea is that the field can be user as a regular one (users can set it) and this mixin is meant for glue modules that would need this same behavior.

For testing part, I gues it could be done somehow, I should look for a runtime model patching or something like that...

Reply:
OK, got it.

Reply:
`odoo_test_helper` library is for that things, but it's not worth right now I think.

Reply:
Oh, I just added a little test for that coverage :)